src/bp/ui-studio/src/web/views/FlowBuilder/index.tsx

```diff
@@ -87,7 +91,34 @@ class FlowBuilder extends Component<Props, State> {
         onDismiss: this.props.clearFlowsModification
       })
     }
+
+    if (!_.isEqual(prevProps.flowsByName, this.props.flowsByName)) {
+      const currentFlow = this.props.flowsByName[this.props.currentFlow]
+      const { currentMutex } = (currentFlow || {}) as any
+      if (currentMutex) {
+        const { lastModifiedBy, remainingSeconds } = currentMutex
+
+        const lockFlow = lastModifiedBy !== this.props.user.email && !!remainingSeconds
+
+        if (!this.state.readOnly && lockFlow) {
+          setTimeout(this.unfreezeFlows(this.props.currentFlow), remainingSeconds * 1000 * MUTEX_UNLOCK_SECURITY_FACTOR)
```

Comment:
You need to keep the timeout ID and if you receive a new update, you clear the old timeout.  Otherwise you have a potential race condition where the old setTimeout will clear the lock (but you actually need to keep the freeze because it changed since then)